src/VisualStudio/Core/Def/Implementation/ProjectSystem/IVisualStudioWorkspaceHost2.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    // TODO: Find a better name for this
+    internal interface IVisualStudioWorkspaceHost2
+    {
+        void OnIsComplete(ProjectId projectId, bool succeeded);
```

Comment:
well, @jasonmalinowski suggested the opposite way. saying we might combine this will multiple failure case.